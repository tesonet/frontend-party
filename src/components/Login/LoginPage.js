import React from 'react';
import LoginForm from './LoginForm';

const LoginPage = () => {
  return(
    <div className="login">
      <div className="login__wrap">
      <img className="login__logo" src="./img/logo.svg" alt="testio" title="testio" />
      <LoginForm />
      </div>
    </div>
  );
};

export default LoginPage;
