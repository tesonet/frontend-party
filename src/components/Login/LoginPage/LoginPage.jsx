import React, { Component } from "react";
import ReactCSSTransitionGroup from "react-addons-css-transition-group";

import LoginForm from "../LoginForm/LoginForm";

import styles from "./LoginPage.module";

export default class LoginPage extends Component {
  render() {
    return (
      <div className={styles.loginContainer} role="main">
        <ReactCSSTransitionGroup
          transitionAppear={true}
          transitionAppearTimeout={600}
          transitionEnterTimeout={600}
          transitionLeaveTimeout={200}
          transitionName={
            this.props.match.path === "/" ? "SlideIn" : "SlideOut"
          }
        >
          <LoginForm history={this.props.history} />
        </ReactCSSTransitionGroup>
      </div>
    );
  }
}
