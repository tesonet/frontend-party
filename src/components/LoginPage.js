import React, { Component } from 'react';
import LoginBox from './LoginBox';
import { Redirect } from 'react-router-dom';
import BEM from './BEM';

import '../login.css'; // webpack does not handle scss files! use OS tools

class LoginPage extends Component {

    constructor(props) {
        super(props);

        var token = localStorage.getItem("token");

        if (token) {
            this.state = { authorized: true }
        } else {
            this.state = { authorized: false }
        }
    }

    render() {
        const redirect  = this.state.authorized;

        if (redirect) {
            return <Redirect to='/cms/server-list'/>;
        }

        return <BEM name="login-page">
            <LoginBox />
            <div className="overlay overlay--branded"></div>
        </BEM>
    }
}

export default LoginPage; 