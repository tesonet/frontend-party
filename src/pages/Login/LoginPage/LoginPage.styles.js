import styled from 'styled-components';
import backgroundImage from '@/assets/images/background-image.png';

export const StyledLoginWrapper = styled.div`
  background-image: url(${backgroundImage});
  width: 100%;
  height: 100%;
  position: fixed; 
  top: 0; 
  left: 0; 
  background-size: cover;
  display: flex;
  align-content: center;
  justify-content: center;
  align-items: center;
`;

export const StyledLoginContainer = styled.div`
  display: flex;
  align-items: center;
  width: 360px;
  flex-direction: column;
`;

export const StyledLogo = styled.img`
  width: 246px;
  height: 64px;
  margin-bottom: 50px;
`;
