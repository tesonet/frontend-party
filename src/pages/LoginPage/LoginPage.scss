.main-login-container {
    background: rgba(11, 15, 39, 0.9) url("../../assets/background-image.png") no-repeat center center fixed;
    background-size: cover;
    display: flex;
    justify-content: space-around;
    align-items: center;
    min-width: 100vw;
    min-height: 100vh;
    height: 1vh; // for IE 11
}
