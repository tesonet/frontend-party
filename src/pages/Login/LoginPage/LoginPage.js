import React from 'react';
import mainLogo from '@/assets/images/logo.svg';
import { StyledLoginWrapper, StyledLoginContainer, StyledLogo } from './LoginPage.styles';
import LoginForm from '../LoginForm/LoginForm';

const LoginPage = () => (
  <StyledLoginWrapper>
    <StyledLoginContainer>
      <StyledLogo src={mainLogo} alt="logo" />
      <LoginForm />
    </StyledLoginContainer>
  </StyledLoginWrapper>
);

export default LoginPage;
