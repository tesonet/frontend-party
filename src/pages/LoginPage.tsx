import * as React from 'react';

import { LoginLayout } from '@layouts/LoginLayout';
import { LoginContainer } from '@containers/Login/Login';

const LoginPage = () => (
  <LoginLayout>
    <LoginContainer />
  </LoginLayout>
);

export default LoginPage;
