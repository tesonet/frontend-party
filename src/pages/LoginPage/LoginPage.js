import React from 'react';
import LoginForm from '../../components/LoginForm';
import './LoginPage.scss';

const LoginPage = () => (
  <div className="main-login-container">
    <LoginForm />
  </div>
);

export default LoginPage;
