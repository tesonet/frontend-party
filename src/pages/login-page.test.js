import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";

import LoginPage from "./login-page";


const component = shallow(<LoginPage/>);

it('renders correctly', () => {
    expect(toJson(component)).toMatchSnapshot();
});
