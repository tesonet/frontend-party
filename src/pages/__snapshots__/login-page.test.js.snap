// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.img
    alt="testio logo"
    src=""
  />
  <Connect(LoginForm) />
</styled.div>
`;
